### DISCUSSION TITLE
Feature Request - Don't allow window to be larger or out of bounds of `root`, also update size/position when browser window is resized

### DISCUSSION THREAD

**Discussion by kodie** [nextapps-de/winbox#138] [Ideas]
So I've been working on a very similar JavaScript library as this one but this one is so much further along and just way better in general.

A feature I think would be extremely handy and something I had implemented in my library is the ability to have the window automatically move/resize if the browser window is resized.

For example:

```js
new WinBox({
  container: 'viewport',
  top: 50,
  left: 1024,
  width: 600,
  height: 600
})
```

So in this case, even though I have `left` set to 1024, if it makes the window overflow out of the viewport, it would modify the `left` attribute to make the window completely visible in the viewport, and if the viewport is less than 600px wide (what the window's `width` attribute is set to), it would automatically modify the `width` attribute to make it fit. Same principle applies to `top` and `height` attributes. Also if it's set to `viewport`, when the user scrolls the size and position would also be automatically changed.

`container` could also be set to another element such as `document.body`.

I saw that you mentioned something about implementing a plugins API to extend the winbox functionality, so I'm not sure if you'd want this kind of functionality implemented into winbox core or have it implemented as a plugin but I'd be happy to implement it either way.

Thank you!

**Comment by kodie** [comment]
So after working on the library a little I realise that we already have the `root` option which functions basically the same as the `container` option I was talking about.

However, you're still able to create windows that larger than the `root` option, so I think maybe we should change that where if `width`, `height`, `x`, or `y`, make the window overflow out the the `root` option, the window should get resized/moved to fit.

Also the window position and size should get updated when the browser window is resized. The `breakpoints` option I mentioned in #72 would also come in handy if this was implemented however I think the `breakpoints` option could probably be made into a WinBox plugin.

**Comment by gregroyal** [comment]
@kodie did you come up with a workaround for this? I just ran into this issue too

**Comment by kodie** [comment]
@gregroyal I did not. This would require a modification to the package and a pull request.

**Reply by cawoodm** [reply]
Why not just listen for your browser window resize event and then move your own winbox windows into the visible are?
